### PR TITLE
apply logger instance to request

### PR DESCRIPTION
**Pull request by JoeChapman**
So we can use the logger instance in any method invoked with a request object

### CODE REVIEW

**Comment by lennym** [comment]
Churchill will automatically add a logger to the request as it is. So the code to overwrite `req.logger` is actually redundant, it's already there.

**Comment by easternbloc** [comment]
word

**Comment by JoeChapman** [comment]
@lennym @easternbloc can this be merged?

**Comment by lennym** [comment]
Setting `LOG_LEVEL=silent` for tests seems like a better approach than hacking everywhere with proxyquire.

**Comment by JoeChapman** [comment]
Using `proxyquire` because `logger` is invoked on each request and throwing 
```
Error: Transport already attached: console
      at Logger.add (node_modules/winston/lib/winston/logger.js:454:11)
      at node_modules/winston/lib/winston/logger.js:62:12
      at Array.forEach (native)
      at new exports.Logger (node_modules/winston/lib/winston/logger.js:61:24)
      at module.exports.config (lib/logger.js:9:1540)
      at bootstrap (index.js:9:6128)
      at Context.it (test/integration/behaviours.js:102:7)
      at Context.fn (node_modules/sinomocha/index.js:151:27)
      at Test.mocha.Test.run (node_modules/sinomocha/index.js:156:27)
```

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   },
   "scripts": {
     "test": "npm run unit && npm run lint",
-    "unit": "NODE_ENV=test istanbul cover _mocha",
+    "unit": "LOG_LEVEL=error NODE_ENV=test istanbul cover _mocha",
     "unit:nocov": "NODE_ENV=test mocha",
```
Comment (lennym) [inline package.json]: We probably want the logs silenced for the `nocov` script too. And I don't think `NODE_ENV` needs to be set now.

**Comment by lennym** [comment]
Can you bump `churchill` to latest please?

**Comment by easternbloc** [comment]
@JoeChapman `1.1.1`

**Pull request merged**